source/scop.c

```diff
@@ -92,3 +97,174 @@ void candl_scop_usr_cleanup(osl_scop_p scop) {
     scop = scop->next;
   }
 }
+
+osl_scop_p candl_scop_remove_unions(osl_scop_p scop) {
```

Comment:
All the functions you've added doesn't have doxygen comment.

And maybe add some comments inside this function explaining what your code is doing (roughly).